core/query/index.go

```diff
@@ -96,13 +108,34 @@ func (ind *Indexer) insertAnnotatedTxs(ctx context.Context, b *bc.Block) ([]*Ann
 		SELECT $1, unnest($2::integer[]), unnest($3::bytea[]), unnest($4::jsonb[])
 		ON CONFLICT (block_height, tx_pos) DO NOTHING;
 	`
-	_, err := ind.db.Exec(ctx, insertQ, b.Height, positions, hashes, annotatedTxs)
+	_, err = ind.db.Exec(ctx, insertQ, b.Height, positions, hashes, annotatedTxs)
 	if err != nil {
 		return nil, errors.Wrap(err, "inserting annotated_txs to db")
 	}
 	return annotatedTxsDecoded, nil
 }
 
+func (ind *Indexer) loadOutpoints(ctx context.Context, outputIDs pq.ByteaArray) (map[bc.OutputID]*bc.Outpoint, error) {
```

Comment:
nit: since bc.Outpoint is not very large, maybe avoid the pointer? you can still check existence of an entry using

```go
outpoint, ok := outpoints[prevoutID]
if ok {
        // ...
}
```